crates/typed-store/src/traits.rs

```diff
@@ -48,6 +48,9 @@ where
     /// Removes every key-value pair from the map.
     fn unsafe_clear(&self) -> Result<(), Self::Error>;
 
+    /// Removes every key-value pair from the map by deleting the underlying file.
```

Comment:
Consider also document what happens if the begin and/or end falls in the middle of a file.

Reply:
Added more documentation about the behavior